java-client/src/test/java/org/opensearch/client/transport/rest_client/RestClientTransportTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.transport.rest_client;
+
+import org.apache.hc.core5.http.HttpHost;
+import org.apache.hc.core5.http.HttpStatus;
+import org.apache.hc.core5.http.ProtocolVersion;
+import org.apache.hc.core5.http.message.RequestLine;
+import org.apache.hc.core5.http.message.StatusLine;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.json.JsonpMapper;
+import org.opensearch.client.opensearch.cluster.GetClusterSettingsRequest;
+import org.opensearch.client.opensearch.cluster.GetClusterSettingsResponse;
+import org.opensearch.client.transport.TransportException;
+import org.opensearch.client.transport.TransportOptions;
+import org.opensearch.client.transport.endpoints.SimpleEndpoint;
+import software.amazon.awssdk.http.SdkHttpMethod;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+public class RestClientTransportTest {
```

Comment:
We don't need to mock that: the integration tests (fe AbstractRequestIT & co) do run against OS clusters w/o security, it is easy to trigger 401 by supplying invalid credentials (and probably even 403 by doing non-permissive action). Please let me know if you need help crafting these test cases.

Reply:
@reta Could you give me an example on the fastest way of instantiating OpenSearch Client with custom credential within AbstractRequestIT. Also could you provide an example of non-permissive action? I was thinking of wrong http method but since it is high-level API I would not be able to construct that

Reply:
Sure, will do that shortly

Reply:
@chenqi0805 here is a quick example to give you an idea, for `org.opensearch.client.opensearch.integTest.restclient.RequestIT` test suite:

```
@Test
    public void testUnauthorized() throws Exception {
        // The security plugin is required
        Assume.assumeThat(isHttps(), equalTo(true));

        final RestClient restClient = RestClient
            .builder(getClusterHosts().toArray(new HttpHost[0]))
            .setHttpClientConfigCallback(httpClientBuilder -> {
                try {
                    final TlsStrategy tlsStrategy = ClientTlsStrategyBuilder
                        .create()
                        .setSslContext(SSLContextBuilder.create().loadTrustMaterial(null, (chains, authType) -> true).build())
                        .setHostnameVerifier(NoopHostnameVerifier.INSTANCE)
                        .setTlsDetailsFactory(new Factory<SSLEngine, TlsDetails>() {
                            @Override
                            public TlsDetails create(final SSLEngine sslEngine) {
                                return new TlsDetails(sslEngine.getSession(), sslEngine.getApplicationProtocol());
                            }
                        })
                        .build();
    
                    final PoolingAsyncClientConnectionManager connectionManager = PoolingAsyncClientConnectionManagerBuilder.create()
                        .setTlsStrategy(tlsStrategy)
                        .build();
    
                    return httpClientBuilder.setConnectionManager(connectionManager);
                } catch (Exception e) {
                    throw new RuntimeException(e);
                }
            })
            .build();

        final RestClientTransport transport = new RestClientTransport(restClient, new JacksonJsonpMapper());
        final OpenSearchClient client = new OpenSearchClient(transport);
        client.nodes().info();
    }
```

When run against secured OS server (ether dockerized or not), it fails on `main` (without your changes) with:

```
jakarta.json.stream.JsonParsingException: Jackson exception: Unrecognized token 'Unauthorized': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (ByteArrayInputStream); line: 1, column: 13]
	at __randomizedtesting.SeedInfo.seed([2A1A3F446909C3E8:B38154320C47377F]:0)
	at org.opensearch.client.json.jackson.JacksonJsonpParser.convertException(JacksonJsonpParser.java:102)
...
```